<?xml version="1.0"?>
<launch>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="reconfigurator"/>
    
    <arg name="camInfo" default="/usb_cam/camera_info" doc="camera_info topic"/>
    <arg name="camRaw" default="/usb_cam/image_raw" doc="image_raw topic"/>

    <arg name="idBits" default="5" doc="Number of encoded bits"/>
    <arg name="idSamples" default="360" doc="Number of samples to detect ID"/>
    <arg name="hammingDist" default="1" doc="Hamming distance used for ID"/>
    <arg name="maxPatterns" default="50" doc="Maximum possible amount of patterns to be tracked"/>

    <arg name="useGui" default="true" doc="Whether to use GUI"/>

    <arg name="fontPath" value="$(find whycon_ros)/etc/font.ttf"/>
    <arg name="calibDefPath" value="$(find whycon_ros)/etc/default.cal"/>
    
    <node pkg="whycon_ros" type="whycon_ros" name="whycon_ros" output="screen" args="$(arg fontPath) $(arg calibDefPath)" launch-prefix="">
        
	<remap from="/camera/camera_info" to="$(arg camInfo)"/>
        <remap from="/camera/image_raw" to="$(arg camRaw)"/>
	
	<param name="useGui" value="$(arg useGui)"/>
	<param name="idBits" value="$(arg idBits)"/>
	<param name="idSamples" value="$(arg idSamples)"/>
        <param name="hammingDist" value="$(arg hammingDist)"/>
        <param name="maxPatterns" value="$(arg maxPatterns)"/>
        
    </node>

</launch>
